src/modules/ctc/src/ctc.cc

```diff
@@ -143,14 +194,20 @@ std::vector<types::TrackCircuitData> Ctc::GetSuggestedSpeedsAndAuthorities(void)
 /*------------------------------------- Setters -------------------------------------*/
 void Ctc::SetBlocks(std::vector<types::Block> &blocks)
 {
-    types::Block yard;
-    yard.block = 0;
-    blocks_.push_back(yard);
-    for (types::Block &block : blocks)
+    if (blocks[CTC_FIRST_BLOCK].track == types::TrackId::TRACKID_GREEN)
     {
-        blocks_.push_back(block);
+        types::Block yard;
+        yard.block = 0;
+        yard.track = types::TrackId::TRACKID_GREEN;
+        blocks_.push_back(yard);
+        for (types::Block &block : blocks)
```

Comment:
~~Move the `for` loop outside the if/else statement so it only has to be written once instead of duplicated for each line.~~
See next comment about not needing an if/else statement.

Reply:
ye